apps/condo/domains/meter/tasks/sendVerificationDateReminder.js

```diff
@@ -0,0 +1,241 @@
+const dayjs = require('dayjs')
+const { getSchemaCtx } = require('@core/keystone/schema')
+const { Meter } = require('@condo/domains/meter/utils/serverSchema')
+const { sendMessage, Message } = require('@condo/domains/notification/utils/serverSchema')
+const { Organization } = require('@condo/domains/organization/utils/serverSchema')
+const { Resident, ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+const { get, flatten, uniq } = require('lodash')
+const { COUNTRIES, DEFAULT_LOCALE } = require('@condo/domains/common/constants/countries')
+const { METER_VERIFICATION_DATE_REMINDER_TYPE } = require('@condo/domains/notification/constants/constants')
+
+const readMetersPage = async ({ context, offset, pageSize, date, searchWindowDaysShift, daysCount }) => {
+    // calc window
+    const startWindowDate = dayjs(date).add(searchWindowDaysShift, 'day').format('YYYY-MM-DD')
+    const endWindowDate = dayjs(date)
+        .add(searchWindowDaysShift + daysCount + 1, 'day')
+        .format('YYYY-MM-DD')
+
+    return await Meter.getAll(
+        context, {
+            nextVerificationDate_gte: startWindowDate,
+            nextVerificationDate_lte: endWindowDate,
+        }, {
+            sortBy: 'id_ASC',
+            first: pageSize,
+            skip: offset,
+        }
+    )
+}
+
+const getMetersConnectedWithResidents = async ({ context, meters }) => {
+    // first step is get service consumers by accountNumbers
+    const accountNumbers = uniq(meters.map(meter => meter.accountNumber))
+    const servicesConsumers = await loadListByChunks({
+        context,
+        list: ServiceConsumer,
+        where: {
+            accountNumber_in: accountNumbers,
+        },
+    })
+
+    // second step is to get all resident ids
+    const residentsIds = uniq(servicesConsumers.map(item => item.resident.id))
+
+    // now let's get residents for the list of ids
+    const residents = await loadListByChunks({
+        context,
+        list: Resident,
+        where: {
+            id_in: residentsIds,
+        },
+    })
+
+    // next step - connect residents to services consumers
+    const servicesConsumerWithConnectedResidents = servicesConsumers.map(servicesConsumer => {
+        return {
+            servicesConsumer,
+            residents: residents.filter(resident => resident.id === servicesConsumer.resident.id),
+        }
+    })
+
+    // next step - connect meters to residents using previously created connection
+    return meters
+        .map(meter => {
+            const servicesConsumers = servicesConsumerWithConnectedResidents
+                .filter(item => item.servicesConsumer.accountNumber === meter.accountNumber)
+            const residents = flatten(
+                servicesConsumers.map(item =>
+                    item.residents.filter(resident =>
+                        resident.unitName === meter.unitName
```

Comment:
unitType!

Reply:
As discussed - there are no unit type at this moment in the prod database